src/dune_config/config.mli

```diff
@@ -0,0 +1,15 @@
+(** General dune configuration library for non user facing configuration.
+
+    The configuration available through this module is non user facing and is
+    subject to change without warning. *)
+
+type 'a t
+
+(** [get t] return the value of the configuration for [t] *)
+val get : 'a t -> 'a
+
+(** should dune acquire the global lock before building *)
+val global_lock : [ `Enabled | `Disabled ] t
+
+(** Before any configuration value is accessed, this function must be called *)
+val init : unit -> unit
```

Comment:
Would it make sense to have this type be:

```
val init : unit -> t
```

? 

It would require passing config through functions, but it'd be more future proof and explicit than having the side effect hidden.

Reply:
For some of the places we want to use this it would be a bit too intrusive. Plus, many of these experimental options are not going to be permanent, so it would be a shame to mangle the code just to accommodate them.